src/test/java/seedu/address/model/employee/ContainsAllPredicateTest.java

```diff
@@ -1,75 +1,181 @@
+        Set<Phone> phoneSet = new CustomSet<>();
+        Set<Email> emailSet = new CustomSet<>();
+        Set<Address> addressSet = new CustomSet<>();
+        Set<Salary> salarySet = new CustomSet<>();
+        Set<Leave> leaveSet = new CustomSet<>();
+        Set<Role> roleSet = new CustomSet<>();
+        Set<EmployeeName> supervisorSet = new CustomSet<>();
+        Set<DepartmentName> departmentSet = new CustomSet<>();
+
+        String[] prefix = prefixes.split("_");
+        String[] keyword = keywords.split("_");
+        for (int i = 0; i < prefix.length; i++) {
+            switch (prefix[i]) {
+            case "name":
+                nameSet.add(new EmployeeName(keyword[i]));
+                break;
+            case "phone":
+                phoneSet.add(new Phone(keyword[i]));
+                break;
+            case "email":
+                emailSet.add(new Email(keyword[i]));
+                break;
+            case "address":
+                addressSet.add(new Address(keyword[i]));
+                break;
+            case "salary":
+                salarySet.add(new Salary(keyword[i]));
+                break;
+            case "leave":
+                leaveSet.add(new Leave(keyword[i]));
+                break;
+            case "role":
+                roleSet.add(new Role(keyword[i]));
+                break;
+            case "supervisor":
+                supervisorSet.add(new EmployeeName(keyword[i]));
+                break;
+            case "department":
+                departmentSet.add(new DepartmentName(keyword[i]));
+                break;
+            default:
+                break;
+            }
+        }
+
+        ContainsAllPredicate predicate = new ContainsAllPredicate(nameSet, phoneSet, emailSet, addressSet, salarySet,
+                leaveSet, roleSet, supervisorSet, departmentSet);
+        return predicate;
+    }
+
+    @Test
+    public void test_containsName_returnsTrue() {
+        String keyword = VALID_NAME_AMY;
+        ContainsAllPredicate predicate = createContainsAllPredicateHelper("name", keyword);
+        Employee employee = new EmployeeBuilder().withName(keyword).build();
+        assertTrue(predicate.test(employee));
+    }
+
+    @Test
+    public void test_doesNotContainName_returnsFalse() {
+        String keyword = VALID_NAME_AMY;
+        ContainsAllPredicate predicate = createContainsAllPredicateHelper("name", keyword);
+        Employee employee = new EmployeeBuilder().withName(VALID_NAME_BOB).build();
+        assertFalse(predicate.test(employee));
+    }
+
+    @Test
+    public void test_containsSalary_returnsTrue() {
+        String keyword = VALID_SALARY_AMY;
+        ContainsAllPredicate predicate = createContainsAllPredicateHelper("salary", keyword);
+        Employee employee = new EmployeeBuilder().withSalary(keyword).build();
+        assertTrue(predicate.test(employee));
+    }
+
     @Test
     public void test_containsAddress_returnsTrue() {
```

Comment:
I don't quite get the removal of these lines?

Reply:
because usage 'createContainsAllPredicateHelper' helper function (line 28) replaces these lines.

'predicate' is created multiple times throughout the test file and the 'createContainsAllPredicateHelper' helper function helps to create the 'predicate'

it fulfils the same purpose and the removed lines. same predicate with singular address attribute.